tests/functional/adapter/test_basic.py

```diff
@@ -41,8 +41,8 @@ class TestGenericTestsSnowflake(BaseGenericTests):
     pass
 
 
-class TestSnapshotCheckColsSnowflake(BaseSnapshotCheckCols):
-    pass
+#class TestSnapshotCheckColsSnowflake(BaseSnapshotCheckCols):
+#    pass
```

Comment:
Ah, ok, this is concerning!

I took a deeper dive in, and I see that the snapshots are always running with `where TRUE` as their `row_changed_expr` [here](https://github.com/dbt-labs/dbt-core/blob/ab0c3510ebe70a67fb669b88e6ea36aff2f46339/core/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L149-L151). That's not right! For `check`-strategy snasphots, this variable should only return `true` if it's `check_cols = 'all'` and a new column was added—neither of which are true for the failing snapshots.

I took a quick look, and the last PR to touch this code was indeed 12 days ago: https://github.com/dbt-labs/dbt-core/pull/4893

We've got a lead, gumshoe!

Reply:
Updates:
- Issue: https://github.com/dbt-labs/dbt-core/issues/5222
- Proposed fix: https://github.com/dbt-labs/dbt-core/pull/5223